tests/TestCase/View/Helper/FormHelperTest.php

```diff
@@ -4481,13 +4481,13 @@ public function testRadio(): void
 
         $result = $this->Form->radio('Model.field', ['option A', 'option B'], ['name' => 'Model[custom]']);
         $expected = [
-            ['input' => ['type' => 'hidden', 'name' => 'Model[custom]', 'value' => '']],
-            ['label' => ['for' => 'model-custom-0']],
-            ['input' => ['type' => 'radio', 'name' => 'Model[custom]', 'value' => '0', 'id' => 'model-custom-0']],
+            ['input' => ['type' => 'hidden', 'name' => 'Model[custom]', 'value' => '', 'id' => 'model-field']],
+            ['label' => ['for' => 'model-field-0']],
+            ['input' => ['type' => 'radio', 'name' => 'Model[custom]', 'value' => '0', 'id' => 'model-field-0']],
             'option A',
             '/label',
-            ['label' => ['for' => 'model-custom-1']],
-            ['input' => ['type' => 'radio', 'name' => 'Model[custom]', 'value' => '1', 'id' => 'model-custom-1']],
+            ['label' => ['for' => 'model-field-1']],
+            ['input' => ['type' => 'radio', 'name' => 'Model[custom]', 'value' => '1', 'id' => 'model-field-1']],
```

Comment:
I still don't think we should be changing these id attributes. It will break userland code.

Reply:
It seems bug. In case custom name attribute, If field id exists then options ids generated as per field id otherwise as per custom name attribute.

Reply:
Maintaining backwards compatibility is more important here. So please limit your changes to just adding `id` attribute to the hidden field. We can discuss changing the way `id` attributes are generated for radio inputs separately if needed.

Reply:
@chetanvarshney Can you update the PR?

Reply:
@othercorey 
I can try, I only added `id` attribute for hidden inputs in radio and multiple checkboxes. 

It seems bug. In case custom name attribute, If field id exists then options ids generated as per field id otherwise as per custom name attribute.

Reply:
@othercorey I have done changes in Form helper and Form test helper, Please review.